proposals/js-promise-integration/Overview.md

```diff
@@ -72,16 +83,34 @@ fetch('demo.wasm').then(response =>
+
+The result is that the `Promise` returned by the `compute_delta` is propagated out immediately to the export and the updated version returned by `update_state`. The update here refers to the capture of the suspended computation.
 
-We can bridge this synchrony gap by using the new JS API.
-In the example, an import of the WebAssembly module is wrapped using `suspender.suspendOnReturnedPromise`, and an export is wrapped using `suspender.returnPromiseOnSuspend`, both using the same `suspender`.
-That `suspender` connects to the two together.
-It makes it so that, if ever the (unwrapped) import returns a Promise, the (wrapped) export returns a Promise, with all the computation in between being "suspended" until the import's Promise resolves.
-The wrapping of the export is essentially adding an `async` marker, and the wrapping of the import is essentially adding an `await` marker, but unlike JavaScript we do not have to explicitly thread `async`/`await` all the way through all the intermediate WebAssembly functions!
+At some point, the original `Promise` created by `compute_delta` will be resolved – when the file `data.txt` has been loaded and parsed. At that point, the suspended computation will be resumed with the value read from the file.
+
+It is important to note that the WebAssembly program itself is not aware of having been suspended. From the perspective of the `update_state` function itself, it called an import, got the result and carried on. There has been no change to the WebAssembly code during this process.
+
+Similarly, we are not changing the normal flow of the JavaScript code: it is not suspended except in the normally expected ways – due to the `Promise` returned by `compute_delta`.
+
+Bracketing the exports and imports like this is strongly analagous to adding an `async` marker to the export, and the wrapping of the import is essentially adding an `await` marker, but unlike JavaScript we do not have to explicitly thread `async`/`await` all the way through all the intermediate WebAssembly functions!
+
+Notice that we did not wrap the `init_state` import, nor did we wrap the exported `get_state` function. These functions will continue to behave as they would normally: `init_state` will return with whatever value the JavaScript code gives it – `2.71` in our case – and `get_state` can be used by any JavaScript code to get the current state.
+
+This example uses a shared `Suspender` object that is fixed before the WebAssembly module itself is created. This is simple to use; but has some disadvantages. The primary limitation is that, because the `Suspender` is fixed, it is not possible to support any form of reentrancy of the WebAssembly module; other than that we have just seen with non-wrapped exports and imports. 
```

Comment:
```suggestion
This example uses a shared `Suspender` object that is fixed before the WebAssembly module itself is created. This is simple to use; but has some disadvantages. The primary limitation is that, because the `Suspender` is fixed, it is not possible to support any form of reentrancy of the WebAssembly module, other than what we have just seen with non-wrapped exports and imports. In particular, only one computation can be "inside" a given `Suspender` at a time—whether suspended or active—which means this approach cannot support multiple concurrent computations within the wrapped WebAssembly exports.
```